src/rpc/server.ts

```diff
@@ -113,16 +122,32 @@ export class Server {
 
             this.metrics.httpRequests.labels(labels).inc()
 
-            const durationMs = reply.getResponseTime()
+            const durationMs = reply.elapsedTime
             const durationSeconds = durationMs / 1000
             this.metrics.httpRequestsDuration
                 .labels(labels)
                 .observe(durationSeconds)
         })
 
-        this.fastify.post("/rpc", this.rpc.bind(this))
-        this.fastify.post("/:version/rpc", this.rpc.bind(this))
-        this.fastify.post("/", this.rpc.bind(this))
+        this.fastify.post("/rpc", this.rpcHttp.bind(this))
+        this.fastify.post("/:version/rpc", this.rpcHttp.bind(this))
+        this.fastify.post("/", this.rpcHttp.bind(this))
+
+        this.fastify.register(async () => {
```

Comment:
can you clarify why do we need this wrapper here pls? as i understand it's something like fastify plugin? but why is it needed here?

Reply:
you mean why do we need `fastify.register`? As I see `.get` shortcut is not implemented for ws handlers

Reply:
Done